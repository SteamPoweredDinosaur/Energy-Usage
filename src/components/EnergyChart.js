import React from 'react';
import { BarChart, Bar, CartesianGrid, XAxis, YAxis, Tooltip, ResponsiveContainer } from 'recharts';
import { DateTime } from 'luxon';
import { monthlyUsageEstimator } from '../usage_estimates';

export default ({ meterReadings }) => {
  const estimator = monthlyUsageEstimator(meterReadings);

  const energyUsageData = [];
  for (let i = 0; i < meterReadings.length - 2; i++) {
    const date = DateTime.fromISO(meterReadings[i + 1].readingDate);

    energyUsageData.push({
      date: DateTime.fromISO(meterReadings[i + 1].readingDate).toFormat('LLL'),
      energyUsage: estimator(date),
    });
  }
  return (
    <ResponsiveContainer height="90%" width="100%" minHeight="100px">
      <BarChart data={energyUsageData}>
        <XAxis dataKey="date" />
        <YAxis dataKey="energyUsage" />
        <CartesianGrid horizontal={false} />
        <Tooltip />
        <Bar dataKey="energyUsage" fill="#03ad54" isAnimationActive={false} />
      </BarChart>
    </ResponsiveContainer>
  );
};
