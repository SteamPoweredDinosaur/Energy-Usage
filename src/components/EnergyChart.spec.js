import React from 'react';
import { shallow } from 'enzyme';
import meterReadings from '../data/meterReadingsSample.json';
import EnergyChart from './EnergyChart';

const setup = (props, render = shallow) => ({ instance: render(<EnergyChart {...props} />), props });

it('renders', () => {
  const { instance } = setup({ meterReadings: meterReadings.electricity });
  expect(instance).toMatchSnapshot();
});
