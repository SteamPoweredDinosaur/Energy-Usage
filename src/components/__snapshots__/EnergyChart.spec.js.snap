// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    meterReadings={
      Array [
        Object {
          "cumulative": 17600,
          "readingDate": "2017-03-31T00:00:00.000Z",
          "unit": "kWh",
        },
        Object {
          "cumulative": 17859,
          "readingDate": "2017-04-30T00:00:00.000Z",
          "unit": "kWh",
        },
        Object {
          "cumulative": 18102,
          "readingDate": "2017-05-31T00:00:00.000Z",
          "unit": "kWh",
        },
        Object {
          "cumulative": 18290,
          "readingDate": "2017-06-30T00:00:00.000Z",
          "unit": "kWh",
        },
        Object {
          "cumulative": 18453,
          "readingDate": "2017-07-31T00:00:00.000Z",
          "unit": "kWh",
        },
        Object {
          "cumulative": 18620,
          "readingDate": "2017-08-31T00:00:00.000Z",
          "unit": "kWh",
        },
        Object {
          "cumulative": 18782,
          "readingDate": "2017-09-30T00:00:00.000Z",
          "unit": "kWh",
        },
        Object {
          "cumulative": 18965,
          "readingDate": "2017-10-31T00:00:00.000Z",
          "unit": "kWh",
        },
        Object {
          "cumulative": 19230,
          "readingDate": "2017-11-30T00:00:00.000Z",
          "unit": "kWh",
        },
        Object {
          "cumulative": 19517,
          "readingDate": "2017-12-31T00:00:00.000Z",
          "unit": "kWh",
        },
        Object {
          "cumulative": 19827,
          "readingDate": "2018-01-31T00:00:00.000Z",
          "unit": "kWh",
        },
        Object {
          "cumulative": 20113,
          "readingDate": "2018-02-28T00:00:00.000Z",
          "unit": "kWh",
        },
        Object {
          "cumulative": 20376,
          "readingDate": "2018-03-31T00:00:00.000Z",
          "unit": "kWh",
        },
        Object {
          "cumulative": 20610,
          "readingDate": "2018-04-30T00:00:00.000Z",
          "unit": "kWh",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <BarChart
        barCategoryGap="10%"
        barGap={4}
        data={
          Array [
            Object {
              "date": "Apr",
              "energyUsage": 258.6537022597731,
            },
            Object {
              "date": "May",
              "energyUsage": 241.71574544709074,
            },
            Object {
              "date": "Jun",
              "energyUsage": 186.56874006948055,
            },
            Object {
              "date": "Jul",
              "energyUsage": 162.68897165658564,
            },
            Object {
              "date": "Aug",
              "energyUsage": 167.2302045431461,
            },
            Object {
              "date": "Sep",
              "energyUsage": 161.78838188942245,
            },
            Object {
              "date": "Oct",
              "energyUsage": 185.4676066264583,
            },
            Object {
              "date": "Nov",
              "energyUsage": 266.82912919117007,
            },
            Object {
              "date": "Dec",
              "energyUsage": 287.1926469801474,
            },
            Object {
              "date": "Jan",
              "energyUsage": 310.9152218601521,
            },
            Object {
              "date": "Feb",
              "energyUsage": 284.9942520343029,
            },
            Object {
              "date": "Mar",
              "energyUsage": 261.141961196965,
            },
          ]
        }
        layout="horizontal"
        margin={
          Object {
            "bottom": 5,
            "left": 5,
            "right": 5,
            "top": 5,
          }
        }
        reverseStackOrder={false}
        stackOffset="none"
      >
        <XAxis
          allowDataOverflow={false}
          allowDecimals={true}
          allowDuplicatedCategory={true}
          dataKey="date"
          domain={
            Array [
              0,
              "auto",
            ]
          }
          height={30}
          hide={false}
          mirror={false}
          orientation="bottom"
          padding={
            Object {
              "left": 0,
              "right": 0,
            }
          }
          reversed={false}
          scale="auto"
          tickCount={5}
          type="category"
          width={0}
          xAxisId={0}
        />
        <YAxis
          allowDataOverflow={false}
          allowDecimals={true}
          allowDuplicatedCategory={true}
          dataKey="energyUsage"
          domain={
            Array [
              0,
              "auto",
            ]
          }
          height={0}
          hide={false}
          mirror={false}
          orientation="left"
          padding={
            Object {
              "bottom": 0,
              "top": 0,
            }
          }
          reversed={false}
          scale="auto"
          tickCount={5}
          type="number"
          width={60}
          yAxisId={0}
        />
        <CartesianGrid
          fill="none"
          horizontal={false}
          horizontalFill={Array []}
          horizontalPoints={Array []}
          stroke="#ccc"
          vertical={true}
          verticalFill={Array []}
          verticalPoints={Array []}
        />
        <Tooltip
          active={false}
          animationDuration={400}
          animationEasing="ease"
          coordinate={
            Object {
              "x": 0,
              "y": 0,
            }
          }
          cursor={true}
          cursorStyle={Object {}}
          filterNull={true}
          isAnimationActive={true}
          itemSorter={[Function]}
          itemStyle={Object {}}
          labelStyle={Object {}}
          offset={10}
          separator=" : "
          useTranslate3d={false}
          viewBox={
            Object {
              "x1": 0,
              "x2": 0,
              "y1": 0,
              "y2": 0,
            }
          }
          wrapperStyle={Object {}}
        />
        <Bar
          animationBegin={0}
          animationDuration={400}
          animationEasing="ease"
          data={Array []}
          dataKey="energyUsage"
          fill="#03ad54"
          hide={false}
          isAnimationActive={false}
          layout="vertical"
          legendType="rect"
          minPointSize={0}
          onAnimationEnd={[Function]}
          onAnimationStart={[Function]}
          xAxisId={0}
          yAxisId={0}
        />
      </BarChart>,
      "debounce": 0,
      "height": "90%",
      "minHeight": "100px",
      "width": "100%",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "barCategoryGap": "10%",
        "barGap": 4,
        "children": Array [
          <XAxis
            allowDataOverflow={false}
            allowDecimals={true}
            allowDuplicatedCategory={true}
            dataKey="date"
            domain={
              Array [
                0,
                "auto",
              ]
            }
            height={30}
            hide={false}
            mirror={false}
            orientation="bottom"
            padding={
              Object {
                "left": 0,
                "right": 0,
              }
            }
            reversed={false}
            scale="auto"
            tickCount={5}
            type="category"
            width={0}
            xAxisId={0}
          />,
          <YAxis
            allowDataOverflow={false}
            allowDecimals={true}
            allowDuplicatedCategory={true}
            dataKey="energyUsage"
            domain={
              Array [
                0,
                "auto",
              ]
            }
            height={0}
            hide={false}
            mirror={false}
            orientation="left"
            padding={
              Object {
                "bottom": 0,
                "top": 0,
              }
            }
            reversed={false}
            scale="auto"
            tickCount={5}
            type="number"
            width={60}
            yAxisId={0}
          />,
          <CartesianGrid
            fill="none"
            horizontal={false}
            horizontalFill={Array []}
            horizontalPoints={Array []}
            stroke="#ccc"
            vertical={true}
            verticalFill={Array []}
            verticalPoints={Array []}
          />,
          <Tooltip
            active={false}
            animationDuration={400}
            animationEasing="ease"
            coordinate={
              Object {
                "x": 0,
                "y": 0,
              }
            }
            cursor={true}
            cursorStyle={Object {}}
            filterNull={true}
            isAnimationActive={true}
            itemSorter={[Function]}
            itemStyle={Object {}}
            labelStyle={Object {}}
            offset={10}
            separator=" : "
            useTranslate3d={false}
            viewBox={
              Object {
                "x1": 0,
                "x2": 0,
                "y1": 0,
                "y2": 0,
              }
            }
            wrapperStyle={Object {}}
          />,
          <Bar
            animationBegin={0}
            animationDuration={400}
            animationEasing="ease"
            data={Array []}
            dataKey="energyUsage"
            fill="#03ad54"
            hide={false}
            isAnimationActive={false}
            layout="vertical"
            legendType="rect"
            minPointSize={0}
            onAnimationEnd={[Function]}
            onAnimationStart={[Function]}
            xAxisId={0}
            yAxisId={0}
          />,
        ],
        "data": Array [
          Object {
            "date": "Apr",
            "energyUsage": 258.6537022597731,
          },
          Object {
            "date": "May",
            "energyUsage": 241.71574544709074,
          },
          Object {
            "date": "Jun",
            "energyUsage": 186.56874006948055,
          },
          Object {
            "date": "Jul",
            "energyUsage": 162.68897165658564,
          },
          Object {
            "date": "Aug",
            "energyUsage": 167.2302045431461,
          },
          Object {
            "date": "Sep",
            "energyUsage": 161.78838188942245,
          },
          Object {
            "date": "Oct",
            "energyUsage": 185.4676066264583,
          },
          Object {
            "date": "Nov",
            "energyUsage": 266.82912919117007,
          },
          Object {
            "date": "Dec",
            "energyUsage": 287.1926469801474,
          },
          Object {
            "date": "Jan",
            "energyUsage": 310.9152218601521,
          },
          Object {
            "date": "Feb",
            "energyUsage": 284.9942520343029,
          },
          Object {
            "date": "Mar",
            "energyUsage": 261.141961196965,
          },
        ],
        "layout": "horizontal",
        "margin": Object {
          "bottom": 5,
          "left": 5,
          "right": 5,
          "top": 5,
        },
        "reverseStackOrder": false,
        "stackOffset": "none",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowDataOverflow": false,
            "allowDecimals": true,
            "allowDuplicatedCategory": true,
            "dataKey": "date",
            "domain": Array [
              0,
              "auto",
            ],
            "height": 30,
            "hide": false,
            "mirror": false,
            "orientation": "bottom",
            "padding": Object {
              "left": 0,
              "right": 0,
            },
            "reversed": false,
            "scale": "auto",
            "tickCount": 5,
            "type": "category",
            "width": 0,
            "xAxisId": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowDataOverflow": false,
            "allowDecimals": true,
            "allowDuplicatedCategory": true,
            "dataKey": "energyUsage",
            "domain": Array [
              0,
              "auto",
            ],
            "height": 0,
            "hide": false,
            "mirror": false,
            "orientation": "left",
            "padding": Object {
              "bottom": 0,
              "top": 0,
            },
            "reversed": false,
            "scale": "auto",
            "tickCount": 5,
            "type": "number",
            "width": 60,
            "yAxisId": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "fill": "none",
            "horizontal": false,
            "horizontalFill": Array [],
            "horizontalPoints": Array [],
            "stroke": "#ccc",
            "vertical": true,
            "verticalFill": Array [],
            "verticalPoints": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "animationDuration": 400,
            "animationEasing": "ease",
            "coordinate": Object {
              "x": 0,
              "y": 0,
            },
            "cursor": true,
            "cursorStyle": Object {},
            "filterNull": true,
            "isAnimationActive": true,
            "itemSorter": [Function],
            "itemStyle": Object {},
            "labelStyle": Object {},
            "offset": 10,
            "separator": " : ",
            "useTranslate3d": false,
            "viewBox": Object {
              "x1": 0,
              "x2": 0,
              "y1": 0,
              "y2": 0,
            },
            "wrapperStyle": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animationBegin": 0,
            "animationDuration": 400,
            "animationEasing": "ease",
            "data": Array [],
            "dataKey": "energyUsage",
            "fill": "#03ad54",
            "hide": false,
            "isAnimationActive": false,
            "layout": "vertical",
            "legendType": "rect",
            "minPointSize": 0,
            "onAnimationEnd": [Function],
            "onAnimationStart": [Function],
            "xAxisId": 0,
            "yAxisId": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <BarChart
          barCategoryGap="10%"
          barGap={4}
          data={
            Array [
              Object {
                "date": "Apr",
                "energyUsage": 258.6537022597731,
              },
              Object {
                "date": "May",
                "energyUsage": 241.71574544709074,
              },
              Object {
                "date": "Jun",
                "energyUsage": 186.56874006948055,
              },
              Object {
                "date": "Jul",
                "energyUsage": 162.68897165658564,
              },
              Object {
                "date": "Aug",
                "energyUsage": 167.2302045431461,
              },
              Object {
                "date": "Sep",
                "energyUsage": 161.78838188942245,
              },
              Object {
                "date": "Oct",
                "energyUsage": 185.4676066264583,
              },
              Object {
                "date": "Nov",
                "energyUsage": 266.82912919117007,
              },
              Object {
                "date": "Dec",
                "energyUsage": 287.1926469801474,
              },
              Object {
                "date": "Jan",
                "energyUsage": 310.9152218601521,
              },
              Object {
                "date": "Feb",
                "energyUsage": 284.9942520343029,
              },
              Object {
                "date": "Mar",
                "energyUsage": 261.141961196965,
              },
            ]
          }
          layout="horizontal"
          margin={
            Object {
              "bottom": 5,
              "left": 5,
              "right": 5,
              "top": 5,
            }
          }
          reverseStackOrder={false}
          stackOffset="none"
        >
          <XAxis
            allowDataOverflow={false}
            allowDecimals={true}
            allowDuplicatedCategory={true}
            dataKey="date"
            domain={
              Array [
                0,
                "auto",
              ]
            }
            height={30}
            hide={false}
            mirror={false}
            orientation="bottom"
            padding={
              Object {
                "left": 0,
                "right": 0,
              }
            }
            reversed={false}
            scale="auto"
            tickCount={5}
            type="category"
            width={0}
            xAxisId={0}
          />
          <YAxis
            allowDataOverflow={false}
            allowDecimals={true}
            allowDuplicatedCategory={true}
            dataKey="energyUsage"
            domain={
              Array [
                0,
                "auto",
              ]
            }
            height={0}
            hide={false}
            mirror={false}
            orientation="left"
            padding={
              Object {
                "bottom": 0,
                "top": 0,
              }
            }
            reversed={false}
            scale="auto"
            tickCount={5}
            type="number"
            width={60}
            yAxisId={0}
          />
          <CartesianGrid
            fill="none"
            horizontal={false}
            horizontalFill={Array []}
            horizontalPoints={Array []}
            stroke="#ccc"
            vertical={true}
            verticalFill={Array []}
            verticalPoints={Array []}
          />
          <Tooltip
            active={false}
            animationDuration={400}
            animationEasing="ease"
            coordinate={
              Object {
                "x": 0,
                "y": 0,
              }
            }
            cursor={true}
            cursorStyle={Object {}}
            filterNull={true}
            isAnimationActive={true}
            itemSorter={[Function]}
            itemStyle={Object {}}
            labelStyle={Object {}}
            offset={10}
            separator=" : "
            useTranslate3d={false}
            viewBox={
              Object {
                "x1": 0,
                "x2": 0,
                "y1": 0,
                "y2": 0,
              }
            }
            wrapperStyle={Object {}}
          />
          <Bar
            animationBegin={0}
            animationDuration={400}
            animationEasing="ease"
            data={Array []}
            dataKey="energyUsage"
            fill="#03ad54"
            hide={false}
            isAnimationActive={false}
            layout="vertical"
            legendType="rect"
            minPointSize={0}
            onAnimationEnd={[Function]}
            onAnimationStart={[Function]}
            xAxisId={0}
            yAxisId={0}
          />
        </BarChart>,
        "debounce": 0,
        "height": "90%",
        "minHeight": "100px",
        "width": "100%",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "barCategoryGap": "10%",
          "barGap": 4,
          "children": Array [
            <XAxis
              allowDataOverflow={false}
              allowDecimals={true}
              allowDuplicatedCategory={true}
              dataKey="date"
              domain={
                Array [
                  0,
                  "auto",
                ]
              }
              height={30}
              hide={false}
              mirror={false}
              orientation="bottom"
              padding={
                Object {
                  "left": 0,
                  "right": 0,
                }
              }
              reversed={false}
              scale="auto"
              tickCount={5}
              type="category"
              width={0}
              xAxisId={0}
            />,
            <YAxis
              allowDataOverflow={false}
              allowDecimals={true}
              allowDuplicatedCategory={true}
              dataKey="energyUsage"
              domain={
                Array [
                  0,
                  "auto",
                ]
              }
              height={0}
              hide={false}
              mirror={false}
              orientation="left"
              padding={
                Object {
                  "bottom": 0,
                  "top": 0,
                }
              }
              reversed={false}
              scale="auto"
              tickCount={5}
              type="number"
              width={60}
              yAxisId={0}
            />,
            <CartesianGrid
              fill="none"
              horizontal={false}
              horizontalFill={Array []}
              horizontalPoints={Array []}
              stroke="#ccc"
              vertical={true}
              verticalFill={Array []}
              verticalPoints={Array []}
            />,
            <Tooltip
              active={false}
              animationDuration={400}
              animationEasing="ease"
              coordinate={
                Object {
                  "x": 0,
                  "y": 0,
                }
              }
              cursor={true}
              cursorStyle={Object {}}
              filterNull={true}
              isAnimationActive={true}
              itemSorter={[Function]}
              itemStyle={Object {}}
              labelStyle={Object {}}
              offset={10}
              separator=" : "
              useTranslate3d={false}
              viewBox={
                Object {
                  "x1": 0,
                  "x2": 0,
                  "y1": 0,
                  "y2": 0,
                }
              }
              wrapperStyle={Object {}}
            />,
            <Bar
              animationBegin={0}
              animationDuration={400}
              animationEasing="ease"
              data={Array []}
              dataKey="energyUsage"
              fill="#03ad54"
              hide={false}
              isAnimationActive={false}
              layout="vertical"
              legendType="rect"
              minPointSize={0}
              onAnimationEnd={[Function]}
              onAnimationStart={[Function]}
              xAxisId={0}
              yAxisId={0}
            />,
          ],
          "data": Array [
            Object {
              "date": "Apr",
              "energyUsage": 258.6537022597731,
            },
            Object {
              "date": "May",
              "energyUsage": 241.71574544709074,
            },
            Object {
              "date": "Jun",
              "energyUsage": 186.56874006948055,
            },
            Object {
              "date": "Jul",
              "energyUsage": 162.68897165658564,
            },
            Object {
              "date": "Aug",
              "energyUsage": 167.2302045431461,
            },
            Object {
              "date": "Sep",
              "energyUsage": 161.78838188942245,
            },
            Object {
              "date": "Oct",
              "energyUsage": 185.4676066264583,
            },
            Object {
              "date": "Nov",
              "energyUsage": 266.82912919117007,
            },
            Object {
              "date": "Dec",
              "energyUsage": 287.1926469801474,
            },
            Object {
              "date": "Jan",
              "energyUsage": 310.9152218601521,
            },
            Object {
              "date": "Feb",
              "energyUsage": 284.9942520343029,
            },
            Object {
              "date": "Mar",
              "energyUsage": 261.141961196965,
            },
          ],
          "layout": "horizontal",
          "margin": Object {
            "bottom": 5,
            "left": 5,
            "right": 5,
            "top": 5,
          },
          "reverseStackOrder": false,
          "stackOffset": "none",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowDataOverflow": false,
              "allowDecimals": true,
              "allowDuplicatedCategory": true,
              "dataKey": "date",
              "domain": Array [
                0,
                "auto",
              ],
              "height": 30,
              "hide": false,
              "mirror": false,
              "orientation": "bottom",
              "padding": Object {
                "left": 0,
                "right": 0,
              },
              "reversed": false,
              "scale": "auto",
              "tickCount": 5,
              "type": "category",
              "width": 0,
              "xAxisId": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowDataOverflow": false,
              "allowDecimals": true,
              "allowDuplicatedCategory": true,
              "dataKey": "energyUsage",
              "domain": Array [
                0,
                "auto",
              ],
              "height": 0,
              "hide": false,
              "mirror": false,
              "orientation": "left",
              "padding": Object {
                "bottom": 0,
                "top": 0,
              },
              "reversed": false,
              "scale": "auto",
              "tickCount": 5,
              "type": "number",
              "width": 60,
              "yAxisId": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "fill": "none",
              "horizontal": false,
              "horizontalFill": Array [],
              "horizontalPoints": Array [],
              "stroke": "#ccc",
              "vertical": true,
              "verticalFill": Array [],
              "verticalPoints": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": false,
              "animationDuration": 400,
              "animationEasing": "ease",
              "coordinate": Object {
                "x": 0,
                "y": 0,
              },
              "cursor": true,
              "cursorStyle": Object {},
              "filterNull": true,
              "isAnimationActive": true,
              "itemSorter": [Function],
              "itemStyle": Object {},
              "labelStyle": Object {},
              "offset": 10,
              "separator": " : ",
              "useTranslate3d": false,
              "viewBox": Object {
                "x1": 0,
                "x2": 0,
                "y1": 0,
                "y2": 0,
              },
              "wrapperStyle": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "animationBegin": 0,
              "animationDuration": 400,
              "animationEasing": "ease",
              "data": Array [],
              "dataKey": "energyUsage",
              "fill": "#03ad54",
              "hide": false,
              "isAnimationActive": false,
              "layout": "vertical",
              "legendType": "rect",
              "minPointSize": 0,
              "onAnimationEnd": [Function],
              "onAnimationStart": [Function],
              "xAxisId": 0,
              "yAxisId": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
